bootstrap/secret/secure.go

```diff
@@ -388,3 +393,24 @@ func (p *SecureProvider) ListSecretPaths() ([]string, error) {
 
 	return secureSecrets, nil
 }
+
+// RegisteredSecretUpdatedCallback registers a callback for a secret.
+func (p *SecureProvider) RegisteredSecretUpdatedCallback(path string, callback func(path string)) {
+	p.registeredSecretCallbacks[path] = callback
+}
+
+// SecretUpdatedAtPath performs updates and callbacks for an updated secret or path.
+func (p *SecureProvider) SecretUpdatedAtPath(path string) {
+	p.lastUpdated = time.Now()
+	if p.registeredSecretCallbacks != nil {
+		// Execute Callback for provided path.
+		for k, v := range p.registeredSecretCallbacks {
+			if k == path {
+				v(path)
+				return
+			}
+		}
+	}
+
+	p.lc.Debugf("no callback registered for path: '%s'", path)
```

Comment:
Print the debug if we're calling the callback.  Don't need debugging for when we're not calling it.